src/plone/restapi/tests/test_services_roles.py

```diff
@@ -30,42 +46,109 @@ def test_roles_endpoint_lists_roles(self):
                 "@type": "role",
                 "id": "Contributor",
                 "title": "Contributor",
+                "can_assign": True,
+            },
+            {
+                "@id": self.portal_url + "/@roles/Editor",
+                "@type": "role",
+                "id": "Editor",
+                "title": "Editor",
+                "can_assign": True,
+            },
+            {
+                "@id": self.portal_url + "/@roles/Member",
+                "@type": "role",
+                "id": "Member",
+                "title": "Member",
+                "can_assign": True,
+            },
+            {
+                "@id": self.portal_url + "/@roles/Reader",
+                "@type": "role",
+                "id": "Reader",
+                "title": "Reader",
+                "can_assign": True,
+            },
+            {
+                "@id": self.portal_url + "/@roles/Reviewer",
+                "@type": "role",
+                "id": "Reviewer",
+                "title": "Reviewer",
+                "can_assign": True,
+            },
+            {
+                "@id": self.portal_url + "/@roles/Site Administrator",
+                "@type": "role",
+                "id": "Site Administrator",
+                "title": "Site Administrator",
+                "can_assign": True,
+            },
+            {
+                "@id": self.portal_url + "/@roles/Manager",
+                "@type": "role",
+                "id": "Manager",
+                "title": "Manager",
+                "can_assign": True,
+            },
+        )
+        result = response.json()
+        self.assertEqual(len(expected), len(result))
+        for item in result:
+            self.assertIn(item, expected)
+
+    def test_siteadm_roles_endpoint_lists_roles(self):
+        self.set_siteadm()
```

Comment:
You may want to use 
setRoles(self.portal, TEST_USER_ID, ["Site Administrator"])

Reply:
@ksuess `self.api_session.get("/@roles")`. `self.api_session` already has the Manger authentication header:

https://github.com/plone/plone.restapi/blob/08f8f871bbf63c4f1033cc3f2c46d371e994aa22/src/plone/restapi/tests/test_services_roles.py#L21

Reply:
OK. I don't have a strong opinion here.